reconcile/ocm_machine_pools.py

```diff
@@ -1,137 +1,332 @@
+        return cls(
+            id=pool.q_id,
+            replicas=pool.replicas,
+            aws_node_pool=AWSNodePool(
+                instance_type=pool.instance_type,
+            ),
+            taints=pool.taints,
+            labels=pool.labels,
+            subnet=pool.subnet,
+            cluster=cluster,
+        )
+
+
+class PoolHandler(BaseModel):
+    action: str
+    pool: AbstractPool
+
+    def act(self, dry_run: bool, ocm: OCM) -> None:
+        logging.info(f"{self.action} {self.pool.dict(by_alias=True)}")
+        if dry_run:
+            return
+
+        if not self.action:
+            pass
+        elif self.action == "delete":
+            self.pool.delete(ocm)
+        elif self.action == "create":
+            self.pool.create(ocm)
+
+
+def fetch_current_state(
+    ocm_map: OCMMap,
+    clusters: Iterable[ClusterV1],
+) -> Mapping[str, Mapping[str, list[AbstractPool]]]:
+    current_state = {}
     for cluster in clusters:
-        cluster_name = cluster["name"]
-        ocm = ocm_map.get(cluster_name)
-        machine_pools = ocm.get_machine_pools(cluster_name)
-        for machine_pool in machine_pools:
-            machine_pool["cluster"] = cluster_name
-            current_state.append(machine_pool)
+        current_state[cluster.name] = {
+            "machine_pools": [],
+        }
+        ocm = ocm_map.get(cluster.name)
+        if cluster.spec.hypershift:
+            machine_pools = ocm.get_node_pools(cluster.name)
+            for machine_pool in machine_pools:
+                current_state[cluster.name]["machine_pools"].append(
+                    NodePool(
+                        id=machine_pool["id"],
+                        replicas=machine_pool["replicas"],
+                        aws_node_pool=AWSNodePool(
+                            instance_type=machine_pool["aws_node_pool"]["instance_type"]
+                        ),
+                        taints=machine_pool.get("taints"),
+                        labels=machine_pool.get("labels"),
+                        subnet=machine_pool.get("subnet"),
+                        cluster=cluster.name,
+                    )
+                )
+        else:
+            machine_pools = ocm.get_machine_pools(cluster.name)
+            for machine_pool in machine_pools:
+                current_state[cluster.name]["machine_pools"].append(
+                    MachinePool(
+                        id=machine_pool["id"],
+                        replicas=machine_pool["replicas"],
+                        instance_type=machine_pool["instance_type"],
+                        taints=machine_pool.get("taints"),
+                        labels=machine_pool.get("labels"),
+                        cluster=cluster.name,
+                    )
+                )
 
-    return ocm_map, current_state
+    return current_state
 
 
-def fetch_desired_state(clusters):
-    desired_state = []
+def fetch_desired_state(
```

Comment:
good opportunity to rename this function since it is not fetching anything, just transforming the given input

Reply:
Good point, renamed to `create_desired_state_from_gql`